### ISSUE TITLE
Multi-value iteration without parens?

### ISSUE THREAD

**Issue by quinnj** [JuliaLang/julia#6484]
Currently, you have to specify parentheses around multiple values when iterators return multiple values (i.e. Dicts, DataFrame columns, etc.). It'd be great if you didn't have to. Was this explicit for some reason? I couldn't find a discussion on it.

``` julia
t = Dict()
t["hey"] = 1
t["ho"] = 2
for (k,v) in t
    println("key=$k, value=$v")
end
#errors
for k,v in t
    println("key=$k, value=$v")
end
```

**Comment by bermanmaxim** [comment]
+1 in favor of this, coming from Python I often expect this to work - mandatory parenthesis appears to me as falsely suggesting something related to tuples

**Comment by yurivish** [comment]
I think this would be really nice to have as well. I always find myself making the mistake, getting a syntax error, then going back to add parentheses.

**Comment by StefanKarpinski** [comment]
Can be done at any point in the 1.x timeframe since it's a non-breaking change.

**Comment by andyDoucette** [comment]
I just ran into this issue.  I think it would be much more intuitive if the destructing syntax in a for loop was the same as outside of the for loop.  Thus, I'd want "for k,v in t" to work just like "k,v =t" works.  Consider this a +1.

**Comment by Keno** [comment]
I don't think we should allow this. I think we're potentially moving in the direction of disallowing this elsewhere because it causes confusing (c.f. https://github.com/JuliaLang/julia/issues/54748).

**Comment by quinnj** [comment]
👍
